### PR TITLE
Issue 1625 - Option for Windows-compatible filenames (was: Filename: Umlaute and special characters)

### PR DESCRIPTION
**Pull request by dgolda**
See: https://tvheadend.org/issues/1625
When filename created by DVR contains both national characters (umlate or ąęłóśź) and special characters (like colon : ), file created by DVR is not accesible from Windows by Samba sharing.
It is possible to remove ALL unsafe characters, but it removes both special characters (it's OK) and national characters (it's not OK).

This pull request adds new option in DVR configuration, called "Replace Samba Unsafe Characters with '_'".
And when this option is checked, replaces unsafe characters / : \ < > | * ? ' " to _

### CODE REVIEW

File: src/dvr/dvr_config.c [hunk 1]
```diff
@@ -727,6 +727,13 @@ const idclass_t dvr_config_class = {
       .off      = offsetof(dvr_config_t, dvr_whitespace_in_title),
       .group    = 5,
     },
+    {
+      .type     = PT_BOOL,
+      .id       = "clean-samba-unsafe",
+      .name     = "Replace Samba Unsafe Characters with '_'",
```
Comment (Jalle19) [inline src/dvr/dvr_config.c]: I would call this "Use Windows-compatible filenames". Normal people don't necessarily know what Samba means.

Comment (dgolda) [inline src/dvr/dvr_config.c]: Could be.

**Comment by dgolda** [comment]
I've changed to caption and variables to suggested "Windows-compatible filenames" and also added description to help.

**Comment by perexg** [comment]
Merged. Thanks.

**Pull request closed without merging**